addons/mail/models/mail_thread.py

```diff
@@ -3547,14 +3546,13 @@ def _notify_thread_by_email(self, message, recipients_data, msg_vals=False,
         #   2. do not send emails immediately if the registry is not loaded,
         #      to prevent sending email during a simple update of the database
         #      using the command-line.
-        test_mode = getattr(threading.current_thread(), 'testing', False)
         if force_send := self.env.context.get('mail_notify_force_send', force_send):
             force_send_limit = int(self.env['ir.config_parameter'].sudo().get_param('mail.mail.force.send.limit', 100))
             force_send = len(emails) < force_send_limit
-        if force_send and (not self.pool._init or test_mode):
+        if force_send and not self.env['ir.mail_server']._disable_send():
```

Comment:
had to read trice because of too much and/or/not conditions, but it seems it is the same :D